coalib/core/CircularDependencyError.py

```diff
@@ -12,8 +13,24 @@ def __init__(self, node=None):
             The node that was encountered twice and closes the dependency
             circle.
         """
-        message = 'Circular dependency detected.'
+        prefix = 'Circular dependency detected'
         if node is not None:
-            message += ' {0!r} -> ... -> {0!r}'.format(node)
+            message = '{prefix}: {node!r} -> ... -> {node!r}'.format(
+                prefix=prefix, node=node)
+        else:
+            message = prefix + '.'
+        return cls(message)
 
-        Exception.__init__(self, message)
+    @classmethod
+    def for_bears(cls, bears):
+        """
+        Creates the CircularDependencyError with a helpful message about the
+        dependency.
+
+        :param bears:
+            sequence of bears, for which a circular dependency was detected.
+        """
+        bear_names = [bear.name for bear in bears]
+
+        return cls('Circular dependency detected: {names}'.format(
+            names=' -> '.join(bear_names)))
```

Comment:
This looks like some performance quirk which doesn't hit. If someone uses an exception, he's likely going to raise it immediately. And this is just saving a little string, imo this doesn't impact performance at all.

Sadly I can't see the comment you're linking to, though if we wanna cache it, we should override `__str__` with `@lru_cache(1)`.